app/models/job_queue.rb

```diff
@@ -104,6 +109,32 @@ def perform_job(job_execution, queue)
     end
   end
 
+  def process_job(job_execution, queue)
+    if staggering_enabled?
+      @stagger_queue.push(job_execution: job_execution, queue: queue)
+    else
+      perform_job(job_execution, queue)
+    end
+  end
+
+  def dequeue_staggered_job
+    @lock.synchronize do
+      perform_job(*@stagger_queue.shift.values) unless @stagger_queue.empty?
+    end
+  end
+
+  def start_staggered_job_dequeuer
+    if staggering_enabled?
```

Comment:
kinda strange that the method says `start this` and then does not do it ... maybe move the if outside ?